internal/desc/service.go

```diff
@@ -0,0 +1,159 @@
+	if path.Path == nil {
+		return nil
+	}
+	return sd.file.srcInfo[pathToString(pathFromSourcePath(path.Path))]
+}
+
+// Unwrap returns the underlying protoreflect.Descriptor.
+func (sd *ServiceDescriptor) Unwrap() protoreflect.Descriptor {
+	return sd.wrapped
+}
+
+// GetServiceOptions returns the service options.
+func (sd *ServiceDescriptor) GetServiceOptions() *descriptorpb.ServiceOptions {
+	return sd.wrapped.Options().(*descriptorpb.ServiceOptions)
+}
+
+// GetMethods returns all methods in the service.
+func (sd *ServiceDescriptor) GetMethods() []*MethodDescriptor {
+	methods := sd.wrapped.Methods()
+	result := make([]*MethodDescriptor, methods.Len())
+	for i := 0; i < methods.Len(); i++ {
+		result[i] = &MethodDescriptor{
+			wrapped: methods.Get(i),
+			file:    sd.file,
+		}
+	}
+	return result
+}
+
+// MethodDescriptor methods
+
+// GetName returns the method name.
+func (md *MethodDescriptor) GetName() string {
+	return string(md.wrapped.Name())
+}
+
+// GetFullyQualifiedName returns the fully qualified method name.
+func (md *MethodDescriptor) GetFullyQualifiedName() string {
+	return string(md.wrapped.FullName())
+}
+
+// GetFile returns the file containing this method.
+func (md *MethodDescriptor) GetFile() *FileDescriptor {
+	return md.file
+}
+
+// GetParent returns the parent service.
+func (md *MethodDescriptor) GetParent() Descriptor {
+	parent := md.wrapped.Parent()
+	if parent == nil {
+		return nil
+	}
+
+	if sd, ok := parent.(protoreflect.ServiceDescriptor); ok {
+		return &ServiceDescriptor{wrapped: sd, file: md.file}
+	}
+	return nil
+}
+
+// GetSourceInfo returns the source code info for this method.
+func (md *MethodDescriptor) GetSourceInfo() *descriptorpb.SourceCodeInfo_Location {
+	path := md.wrapped.ParentFile().SourceLocations().ByDescriptor(md.wrapped)
+	if path.Path == nil {
+		return nil
+	}
+	return md.file.srcInfo[pathToString(pathFromSourcePath(path.Path))]
+}
+
+// Unwrap returns the underlying protoreflect.Descriptor.
+func (md *MethodDescriptor) Unwrap() protoreflect.Descriptor {
+	return md.wrapped
+}
+
+// GetMethodOptions returns the method options.
+func (md *MethodDescriptor) GetMethodOptions() *descriptorpb.MethodOptions {
+	return md.wrapped.Options().(*descriptorpb.MethodOptions)
+}
+
+// GetInputType returns the input message type.
+func (md *MethodDescriptor) GetInputType() *MessageDescriptor {
+	input := md.wrapped.Input()
+	if input == nil {
+		return nil
+	}
+
+	return &MessageDescriptor{
+		wrapped: input,
+		file:    md.file,
+	}
+}
+
+// GetOutputType returns the output message type.
+func (md *MethodDescriptor) GetOutputType() *MessageDescriptor {
+	output := md.wrapped.Output()
+	if output == nil {
+		return nil
+	}
+
+	return &MessageDescriptor{
+		wrapped: output,
+		file:    md.file,
+	}
+}
+
+// IsClientStreaming returns true if this is a client streaming method.
+func (md *MethodDescriptor) IsClientStreaming() bool {
+	return md.wrapped.IsStreamingClient()
```

Comment:
seems like the wrapper should be consistent with the naming of the wrapped object?